### ISSUE TITLE
What happens to un-migrated apps after the April 28th deadline?

### ISSUE THREAD

**Issue by rhymes** [ParsePlatform/parse-server#1639]
I can't find an official information about this. The Wiki page https://github.com/ParsePlatform/parse-server/wiki/Migrating-an-Existing-Parse-App states that the first phase has to be completed before April 28th, which we didn't know about (the blog stated January 28, 2017).

We have a few production apps on Parse.com, and it's likely impossible we are able to migrate their database in two days, so, what's going to happen to existing apps which haven't been migrated?

**Comment by theill** [comment]
As I've understood Parse will prioritize migrated instances after this deadline but will not shut anything down. I read it as your QPS performance might take a hit or something similar. I found this comment from Drew => https://github.com/ParsePlatform/parse-server/issues/1106#issuecomment-203067072 from

**Comment by theill** [comment]
Parse will start to prioritize migrated instances over non-migrated because they see those as more important. At least that's how I've understood it from various comments - such as https://github.com/ParsePlatform/parse-server/issues/1106#issuecomment-203067072 from Drew.

**Comment by rhymes** [comment]
@theill thank you! I opened this because I found it odd there was no official statement about this thing.

**Comment by hramos** [comment]
Reopening this issue since we have been getting multiple duplicate questions about this.

**_tl;dr:_ We don't expect most applications that have not migrated their database by April 28 to see an adverse hit to their performance.**

Our migration guide used to indicate that applications that did not migrate by April 28th would have their traffic de-prioritized. This was poor wording on my part and I apologize for the confusion this has caused.

### We're prioritizing traffic to apps that migrate their database

We want everyone to have a smooth transition, therefore applications that migrate their database will have their traffic to `api.parse.com` handled by a higher priority queue and dedicated machines. We don't foresee this having any major effect on most apps that have not migrated their database yet.

There is a small number of apps that may see degraded performance, and we have already proactively reached out to these developers  multiple times over the past three months. Requests that are affected by this will see an error code 428 in their error logs. If you haven't received any emails requesting action in the email address attached to your Parse App, you don't need to worry yet.

### Why take action now? Don't I have a year to migrate?

Our migration guide recommends updating your client applications to stop using `api.parse.com` by July 28, 2016. By this date you should have either Parse Server or another backend in place to serve your app. That date was chosen as it marks the halfway point since the shutdown was announced and the date when the Parse service will be retired. Releasing an updated app by then will provide you with the best chances of successfully getting most of your user base to update their app before the older version that relies on api.parse.com stops working.

Your new backend will need to be served by a database under your control, therefore migrating your database off Parse is a hard prerequisite. While you can use Parse Server with an empty database, you can't fully test how your app will perform with Parse Server until you migrate your database. Therefore we picked April 28, three months after the announcement and 25% of the way, as our recommended target date to have finished your database migration. This date was picked with developer's best interest in mind as it will give you three months to get Parse Server up and running to your satisfaction.
You won't be able to access your Parse database after Parse is retired on January 28, 2017, either, so migrating your database as soon as possible is the responsible thing to do if you care about your data.

**Comment by hramos** [comment]
Closing as the deadline is now behind us. If you are here because your app is seeing a 428 error code, please migrate your database as soon as possible.

**Comment by 23taps** [comment]
Hector, could you please provide more info on the 428 error code. When parse decided to shut down (which was already bad enough) we were told we would have 1 year to migrate. Now it seems we only had 3 months to migrate our database before getting traffic reprioritized. So first the system is shut down, then we are told we have 1 year, and not we don't even have that? This is beyond unacceptable. And now push notifications are not working and we're getting a 500 error. Were do we report that?

**Comment by georgel** [comment]
Hi,

As a 3+ year customer of Parse, I think the decision to de-prioritize apps that hasn't migrated is flawed.  You are making the assumption that the developer intends to continue running the app on Parse server. Some of us are in the process of moving to an entire new stack. Which makes migrating the DB pointless. 

Can Parse please keep the service running until the announced EOL date at the high quality levels we have enjoyed. As if right now I can't tail the logs to investigate bugs. The console browser is not showing the most recent objects.  I assure you that I can be trusted to manage the migration timeline on my own.

**Comment by hramos** [comment]
The database migration process is completely independent of Parse Server or any other backend service you may decide to use. Whatever you plan on using for your backend, you will need to migrate your data out and into a database under your control _as soon as possible and without further delay_. 

The only exception to this would be if you plan on starting from scratch, on an empty database, and you don't care about the data your app has stored so far. In that case, the migration guide obviously does not apply to you.

The service will stop responding to all requests on January 28th, 2017, and we have no plans of extending that deadline. You are free to follow your own timeline, but we want to stress here that the best experience can be had if you follow our guidelines. 

The next milestone is to start handling requests using Parse Server or another backend by July 28, 2016. These dates have been posted at the top of the migration guide since January 27 of this year.

**Comment by georgel** [comment]
Hi Hramos,

I am planning to move to a new backend and I don't plan to migrate the existing data. I have a migration plan in mind to move my users to the new platform. While I am still working on the new apps, can you please help me restore the same SLA as before?
